### PR TITLE
feat(frontend): new authenticate button with animation

### PR DESCRIPTION
**Pull request by AntonioVentilii-DFINITY**
# Motivation

New authenticate button with animated icon:

https://github.com/user-attachments/assets/51a921f1-d0d5-4ea8-90cc-cb53af06ea8d

### CODE REVIEW

File: src/frontend/src/lib/components/icons/IconArrowAstronaut.svelte [hunk 1]
```diff
@@ -0,0 +1,141 @@
+<svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg" class="button__icon">
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/icons/IconArrowAstronaut.svelte:1]: `<!-- source: DFINITY foundation -->`

File: src/frontend/src/lib/components/icons/IconArrowAstronaut.svelte [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		stroke-dasharray: 20;
+		stroke-dashoffset: 20;
+		transition: stroke-dashoffset 0.4s cubic-bezier(0.3, 0.7, 0, 1);
+	}
+
+	.svg-arrow-go line {
+		transition-delay: 0.6s;
+	}
+
+	button:hover .svg-arrow-go path,
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/icons/IconArrowAstronaut.svelte:115]: So, it works only on desktop. That's the definitive expectation?

File: src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<script>
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte:1]: `lang=ts`

File: src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+	class="h-16 px-8 py-2 rounded-full gap-4 font-bold text-lg"
+	data-tid="login-button"
+>
+	<slot />
+	<IconArrowAstronaut />
+</button>
+
+<style lang="scss">
+	button {
+		background: #016dfc;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte:16]: Is that an existing color `016dfc`, can we use a variable?
If not, should this color be standadized?

File: src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+		background-size: 600% 100%;
+		background-position: 0 0;
+		transition:
+			background-position 0.7s ease-in-out,
+			box-shadow 0.5s ease-in-out;
+	}
+
+	button:hover {
+		background-position: 100% 0;
+		box-shadow: 0 0 50px #1bb5ff;
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte:37]: Same as previous question. A specific color or it should be standardized?

File: src/frontend/src/lib/components/icons/IconArrowAstronaut.svelte (hunk 2 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/icons/IconArrowAstronaut.svelte:115]: for now yes, for what I understood

File: src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte (hunk 3 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte:1]: right!

File: src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte (hunk 4 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte:16]: i am including it in our variables

File: src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte (hunk 5 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/ui/ButtonAuthenticate.svelte:37]: same as before

File: src/frontend/src/lib/components/icons/IconArrowAstronaut.svelte (hunk 1 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/components/icons/IconArrowAstronaut.svelte:1]: Right!

**Review by peterpeterparker — APPROVED** [review]
LGTM, thx

**Comment by AntonioVentilii-DFINITY** [comment]
I forgot to mention that the code for the icon animation was provided by David Aerne

**Pull request merged**